### PR TITLE
Silence `To copy construct from a tensor, it is recommended to use sourceTensor. clone(). detach() or sourceTensor. clone()` warning

### PR DESCRIPTION
**Pull request by ExtReMLapin**
Didn't test it but there is no reason it should not work

### CODE REVIEW

File: python/xgrammar/kernels/apply_token_bitmask_inplace_triton.py [hunk 1]
```diff
@@ -106,8 +106,11 @@ def apply_token_bitmask_inplace_triton(
     num_rows = len(indices) if indices is not None else logits.shape[0] if logits.ndim == 2 else 1
 
     if indices is not None:
-        indices_cpu = torch.tensor(indices, dtype=torch.int32)
-        indices = indices_cpu.to(device=logits.device, non_blocking=True)
+        if isinstance(indices, torch.Tensor):
+            indices = indices.to(dtype=torch.int32, device=logits.device, non_blocking=True)
+        else:
+            indices_cpu = torch.tensor(indices, dtype=torch.int32)
+            indices = indices_cpu.to(device=logits.device, non_blocking=True)
```
[the remark this answers is not in the document]
Comment (Ubospica) [reply python/xgrammar/kernels/apply_token_bitmask_inplace_triton.py:113]: I feel given this is on the critical path of LLM inference, adding extra check is expensive. The provider should check if it's valid, or we can wrap this logic into c++ in the future.

**Comment by ExtReMLapin** [comment]
I’m out of time at the office did not have the time to test

**Comment by ExtReMLapin** [comment]
my point is, look, it's a very simple fix, but Again, I didn't take the 5 required minutes to apply the change, recompile and test it, just to be sure.

**Comment by Ubospica** [comment]
@ExtReMLapin Thanks for the explanation! This looks good enough.

@Seven-Streams Could you check it a bit? We can merge it as long as it runs correctly.
